instrumentor/controllers/deleteinstrumentationconfig/namespace_controller.go

```diff
@@ -165,11 +169,24 @@ func syncGenericWorkloadListToNs(ctx context.Context, c client.Client, kind work
 // when reconciling the namespace, the usecase is to delete instrumentation for workloads that were only
 // instrumented due to the label on the namespace. These are workloads with the label missing.
 // (they inherit the instrumentation from the namespace this way)
-func isInheritingInstrumentationFromNs(obj client.Object) bool {
+func isInheritingInstrumentationFromNs(ctx context.Context, c client.Client, obj client.Object) (bool, error) {
+	sourceList, err := v1alpha1.GetWorkloadSources(ctx, c, obj)
+	if err != nil {
+		return false, err
+	}
+
+	if sourceList.Namespace != nil && sourceList.Namespace.DeletionTimestamp.IsZero() {
+		return true, nil
+	}
+
+	if sourceList.Workload != nil && sourceList.Workload.DeletionTimestamp.IsZero() {
+		return false, nil
+	}
```

Comment:
from previous review - since we first check the namespace and returns `true` if it exists, we will return wrong value here if the `sourceList.Workload` also exists (and then the workload is not inheriting)

consider updating the implementation or rename the function to reflect it's return value)